vault/resource_azure_secret_backend_role.go

```diff
@@ -161,35 +173,35 @@ func azureSecretBackendRoleCreate(d *schema.ResourceData, meta interface{}) erro
 	path := azureSecretRoleResourcePath(backend, role)
 
 	data := map[string]interface{}{}
-	err := azureSecretBackendRoleUpdateFields(d, data)
-	if err != nil {
-		return err
+	diags := azureSecretBackendRoleUpdateFields(ctx, d, data)
```

Comment:
looks like you're only returning `diags` if there is an error; else it is `nil`, in that case we have a style precedent of something like 

```
if diags := azureSecretBackendRoleUpdateFields(ctx, d, data); diags != nil {
    return diags
}
```